src/components/UriViewer.js

```diff
@@ -99,17 +103,33 @@ const PreviewContent = ({ uri, metadata, metadata: { bucket, name }, googleProje
   ])
 }
 
-const DownloadButton = ({ uri, metadata: { bucket, name, fileName, size } }) => {
+const DownloadButton = ({ uri, metadata: { bucket, name, fileName, size }, accessUrl }) => {
   const signal = Utils.useCancellation()
   const [url, setUrl] = useState()
   const getUrl = async () => {
-    try {
-      const { url } = await Ajax(signal).Martha.getSignedUrl({ bucket, object: name, dataObjectUri: isGs(uri) ? undefined : uri })
-      const workspace = workspaceStore.get()
-      const userProject = await getUserProjectForWorkspace(workspace)
-      setUrl(knownBucketRequesterPaysStatuses.get()[bucket] ? Utils.mergeQueryParams({ userProject }, url) : url)
-    } catch (error) {
-      setUrl(null)
+    if (accessUrl && accessUrl.url) {
+      /*
+      NOTE: Not supporting downloading using `accessUrl.headers`:
+      - https://ga4gh.github.io/data-repository-service-schemas/preview/release/drs-1.1.0/docs/#_accessurl
+
+      If we want to support supplying `accessUrl.headers` here we'll probably need a bigger solution.
+      As of 2021-05-17 a google search turned up this c. 2018 result that mentioned something called `ServiceWorker`
+      - https://stackoverflow.com/questions/51721904/make-browser-submit-additional-http-header-if-click-on-hyperlink#answer-51784608
+       */
+      setUrl(_.isEmpty(accessUrl.headers) ? accessUrl.url : null)
+    } else {
+      try {
+        const { url } = await Ajax(signal).Martha.getSignedUrl({
+          bucket,
+          object: name,
+          dataObjectUri: isDrs(uri) && uri
```

Comment:
Nope. This is saying "if this is a DOS URI or a DRS URI, then feed it into the `getSignedUrlV1` field `dataObjectUri`, otherwise leave the `dataObjectUri` as undefined".

We can go over the `getSignedUrlV1` API in more detail and how it needs... help. For now, from a high level, `getSignedUrlV1` \*always\* expects to be passed a GCS bucket and GCS object. `getSignedUrlV1` _optionally_ accepts an additional DOS URI or DRS URI as `dataObjectUri`. If `dataObjectUri` is provided then [the credentials retrieved](https://github.com/broadinstitute/martha/blob/3.6.0/handlers/getSignedUrlV1.js#L9-L11) to generate the GCS signed URL are from one-of-the-Bonds instead of SAM.

Reply:
I think in this case a ternary might be clearer than relying on the short-circuiting, since we actively want `undefined` in the falsy case: `dataObjectUri: isDrs(uri) ? uri : undefined`

Reply:
Used `null` as an explicit value instead. I've been chastised by eslint for daring to explicitly use `undefined` before. Apparently, `undefined` is not a keyword and thus can be shadowed to do unexpected things: https://eslint.org/docs/rules/no-undefined